connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ConnectorConfigTest.java

```diff
@@ -0,0 +1,176 @@
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class ConnectorConfigTest {
+
+    public static abstract class TestConnector extends Connector {
+    }
+
+    public static class SimpleTransformation<R extends ConnectRecord<R>> implements Transformation<R>  {
+
+        int magicNumber = 0;
+
+        @Override
+        public void init(Map<String, Object> props) {
+            magicNumber = Integer.parseInt((String) props.get("magic.number"));
+        }
+
+        @Override
+        public R apply(R record) {
+            return null;
+        }
+
+        @Override
+        public void close() {
+            magicNumber = 0;
+        }
+
+        @Override
+        public ConfigDef config() {
+            return new ConfigDef()
+                    .define("magic.number", ConfigDef.Type.INT, ConfigDef.NO_DEFAULT_VALUE, ConfigDef.Range.atLeast(42), ConfigDef.Importance.HIGH, "");
+        }
+    }
+
+    @Test
+    public void noTransforms() {
+        Map<String, String> props = new HashMap<>();
+        props.put("name", "test");
+        props.put("connector.class", TestConnector.class.getName());
+        new ConnectorConfig(props);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void danglingTransformAlias() {
+        Map<String, String> props = new HashMap<>();
+        props.put("name", "test");
+        props.put("connector.class", TestConnector.class.getName());
+        props.put("transforms", "dangler");
+        new ConnectorConfig(props);
+    }
+
+    @Test
+    public void wrongTransformationType() {
+        Map<String, String> props = new HashMap<>();
+        props.put("name", "test");
+        props.put("connector.class", TestConnector.class.getName());
+        props.put("transforms", "a");
+        props.put("transforms.a.type", "uninstantiable");
+        try {
+            new ConnectorConfig(props);
+            fail();
+        } catch (ConfigException e) {
+            assertTrue(e.getMessage().contains("Class uninstantiable could not be found"));
```

Comment:
Checking error messages is kind of brittle. Do we even need to check this or is it good enough that the test throws an exception?